sygnal/apnspushkin.py

```diff
@@ -250,14 +269,16 @@ async def _dispatch_notification_unlimited(self, n, device, context):
         ) as span_parent:
 
             if n.event_id and not n.type:
-                payload = self._get_payload_event_id_only(n, device)
+                payload: Optional[Dict[str, Any]] = self._get_payload_event_id_only(
+                    n, device
+                )
             else:
                 payload = self._get_payload_full(n, device, log)
 
             if payload is None:
                 # Nothing to do
                 span_parent.log_kv({logs.EVENT: "apns_no_payload"})
-                return
+                return None
```

Comment:
```suggestion
                return []
```
I'm in favour of avoiding confusion in the return type; nothing wrong with returning None but the empty list makes sense here I think